raiden/connection_manager.py

```diff
@@ -309,8 +310,15 @@ def _find_new_partners(self) -> List[Address]:
         )
 
         available_addresses = list(participants_addresses - known)
-        shuffle(available_addresses)
-        new_partners = available_addresses
+        available_and_online_addresses = available_addresses
+        available_and_online_addresses = [
+            address
+            for address in available_addresses
+            if self.raiden.transport.force_check_address_reachability(address)
```

Comment:
This will effectively add all addresses into the `UserAddressManager` and do 2 REST API calls for each, currently we have a few hundred of addresses, which means the connection manager will be a bottle neck.

Please, change this to track the presence of fewer addresses and to do less HTTP requests.

@ulope do we need to limit the addresses which are being tracked to begin with?

Reply:
> Please, change this to track the presence of fewer addresses and to do less HTTP requests.

Thank you for your review!

The way to do this would be to move the forced presence to right before opening the channels, so where the number of channel opens is limited by the user's `n_to_join` setting?

Somewhere here:

https://github.com/raiden-network/raiden/blob/f9bbbc0aeadf2ce1a54f3ed22de948133bc461df/raiden/connection_manager.py#L417-L427

It should be before spawning the greenlets but after having limited the list of potential partners. This way if someone is offline we can still try to see if others from the list are online.

Do you think it should be done this way? If not what would your suggestion be?

Reply:
I don't have any particular suggestion on how to implement it, what you wrote sound correct.

Reply:
Thanks! Want to make sure we are on the same page before implementing anything.

Reply:
done

Reply:
> do we need to limit the addresses which are being tracked to begin with?

Yeah that's a valid question. 

A quick test shows that per 1k addresses with 2 users each the size seems to grow pretty linearly (sampled at 100, 1k, 5k and 20k) by:
- `UserAddressManager` (including the `DisplayNameCache`): 7MB 
- Just the internal state objects: 1MB